web/src/use/use-notifications.ts

```diff
@@ -0,0 +1,69 @@
+import { type Ref, reactive, toRefs, ref, unref, watch } from "vue"
+
+import notificationsApi, {
+  NotificationSourceTypes,
+  type Notification,
+  type NotificationQueryOptions,
+  type NotificationWhereOptions,
+  type NotificationFiltersOptions,
+} from "@/api/notifications-api"
+
+export {
+  NotificationSourceTypes,
+  type Notification,
+  type NotificationQueryOptions,
+  type NotificationWhereOptions,
+  type NotificationFiltersOptions,
+}
+
+export function useNotifications(
+  queryOptions: Ref<NotificationQueryOptions> = ref({}),
+  { skipWatchIf = () => false }: { skipWatchIf?: () => boolean } = {}
+) {
+  const state = reactive<{
+    notifications: Notification[]
+    totalCount: number
+    isLoading: boolean
+    isErrored: boolean
+  }>({
+    notifications: [],
+    totalCount: 0,
+    isLoading: false,
+    isErrored: false,
+  })
+
+  async function fetch(): Promise<Notification[]> {
+    state.isLoading = true
+    try {
+      const { notifications, totalCount } = await notificationsApi.list(unref(queryOptions))
+      state.isErrored = false
+      state.notifications = notifications
+      state.totalCount = totalCount
+      return notifications
+    } catch (error) {
+      console.error("Failed to fetch notifications:", error)
```

Comment:
Consider using the more informative error format of
```suggestion
      console.error(`Failed to fetch notifications: ${error}`, { error })
```
So it's consistent between front- and back-ends (and so we get more error details)